### PR TITLE
goresym 2.6.3 (new formula)

### LINKED ISSUE #44: Build release packages for linux/windows/macOS + add to homebrew package manager (macOS)
The README says:

- https://github.com/mandiant/GoReSym#usage
  - > You can download pre-built `linux` and `windows` GoReSym binaries from the [Releases tab](https://github.com/mandiant/GoReSym/releases/).

Yet looking at the releases page, there doesn't seem to have been windows/linux binary releases since v2.4:

- https://github.com/mandiant/GoReSym/releases/
- https://github.com/mandiant/GoReSym/releases/tag/v2.4

And despite this issue implying this should be able to be built for macOS, I don't see any binary macOS releases:

- https://github.com/mandiant/GoReSym/issues/24

### PR DESCRIPTION
**Pull request by 0xdevalias**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Add new formula for `GoReSym` CLI:

- https://github.com/mandiant/GoReSym
  - > GoReSym is a Go symbol parser that extracts program metadata (such as CPU architecture, OS, endianness, compiler version, etc), function metadata (start & end addresses, names, sources), filename and line number metadata, and embedded structures and types. This cross platform program is based directly on the [open source Go compiler](https://github.com/golang/go/tree/master/src/debug/gosym) and runtime code.
  - fixes https://github.com/mandiant/GoReSym/issues/44#issuecomment-1827128713

### CODE REVIEW

File: Formula/g/goresym.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    ldflags = %w[
+      -s -w
+    ]
+    system "go", "build", *std_go_args(ldflags: ldflags)
+  end
+
+  test do
+    json_output = JSON.parse(shell_output("#{bin}/goresym '#{bin}/goresym'"))
+    assert_equal json_output["BuildInfo"]["Main"]["Path"], "github.com/mandiant/GoReSym"
+  end
```
Comment (0xdevalias) [inline Formula/g/goresym.rb:21]: Note: This test basically uses the `goresym` CLI to parse it's own Golang binary, which will then return json data about it, which it then checks the `["BuildInfo"]["Main"]["Path"]` of to confirm that it is all working properly.

File: Formula/g/goresym.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  license "MIT"
+  head "https://github.com/mandiant/GoReSym.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %w[
+      -s -w
+    ]
+    system "go", "build", *std_go_args(ldflags: ldflags)
```
Comment (chenrui333) [inline Formula/g/goresym.rb:15]:
```suggestion
    ldflags = %w[
      -s -w
    ]
    system "go", "build", *std_go_args(ldflags: "-s -w")
```

File: Formula/g/goresym.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/g/goresym.rb:21]: 👍

**Review by chenrui333 — APPROVED** [review]
LGTM, just one nit comment

**Comment by 0xdevalias** [comment]
> LGTM, just one nit comment

@chenrui333 Accepted your changes :)

**Comment by chenrui333** [comment]
@Oxyriser can you squash all the commits? Thanks!

**Comment by 0xdevalias** [comment]
> can you squash all the commits?

@chenrui333 I've squashed the commits and force pushed.. but honestly that really shouldn't be a blocker in the automation on this repo. PR's can be squash merged, which would solve this and other 'friction points' (like the CI blocking a PR because a certain commit within it is named poorly) without requiring manual process on each and every PR.

**Review by carlocab — APPROVED** [review]

**Comment by chenrui333** [comment]
@0xdevalias thanks for your contribution! 🎉

**Pull request merged**